### PR TITLE
Auth refactor

### PR DESCRIPTION
**Pull request by savonarola**
Fixes [EMQX-9262](https://emqx.atlassian.net/browse/EMQX-9262)

### <samp>🤖 Generated by Copilot at 5c40ea9</samp>

This pull request refactors and integrates the authentication and authorization features of the EMQX broker into a single application called `emqx_auth`. It renames and updates the modules, files, and configuration options related to the authentication and authorization logic. It also improves the code quality and readability of the schema injection and validation functions.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [na] Added property-based tests for code which performs user input validation
- [x] Changed lines (mostly) covered in coverage report, the PR is quite big
- [na] Change log has been added to `changes/(ce|ee)/(feat|perf|fix)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [na] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [x] Schema changes are backward compatible

### CODE REVIEW

File: apps/emqx_auth/src/emqx_authz/emqx_authz.erl [hunk 1]
```diff
@@ -163,7 +186,14 @@ pre_config_update(Path, Cmd, Sources) ->
...
         NSources -> {ok, NSources}
     catch
-        _:Reason -> {error, Reason}
+        Error:Reason:Stack ->
+            ?SLOG(info, #{
+                msg => "error_in_pre_config_update",
+                exception => Error,
+                reason => Reason,
+                stacktrace => Stack
+            }),
```
Comment (zmstone) [inline apps/emqx_auth/src/emqx_authz/emqx_authz.erl:195]: maybe dont log stacktrace when `Error` is `throw`

File: apps/emqx_auth/src/emqx_authz/emqx_authz.erl [hunk 2]
```diff
@@ -566,59 +616,79 @@ take(Type, Sources) ->
...
+    end.
+
+maybe_read_source_files_safe(Source0) ->
+    try maybe_read_source_files(Source0) of
+        Source1 ->
+            {ok, Source1}
+    catch
+        Error:Reason:Stacktrace ->
+            ?SLOG(error, #{
+                msg => "error_in_maybe_read_source_files",
```
Comment (zmstone) [inline apps/emqx_auth/src/emqx_authz/emqx_authz.erl:685]: mfa is logged, there maybe `msg => error_when_reading_source_files`

File: apps/emqx_auth/src/emqx_authz/emqx_authz.erl [hunk 3]
```diff
@@ -502,29 +558,23 @@ changed_paths(OldSources, NewSources) ->
...
-maybe_convert_acl_file(RawConf, _Fun) ->
+maybe_convert_sources(RawConf, _Fun) ->
     RawConf.
 
-read_acl_file(#{<<"path">> := Path} = Source) ->
-    {ok, Rules} = emqx_authz_file:read_file(Path),
-    maps:remove(<<"path">>, Source#{<<"rules">> => Rules}).
+% read_acl_file(#{<<"path">> := Path} = Source) ->
+%     {ok, Rules} = emqx_authz_file:read_file(Path),
+%     maps:remove(<<"path">>, Source#{<<"rules">> => Rules}).
```
Comment (zmstone) [inline apps/emqx_auth/src/emqx_authz/emqx_authz.erl:577]: delete

**Review by zmstone — APPROVED** [review]

**Comment by zmstone** [comment]
This is EPIC. thank you @savonarola

**Comment by savonarola** [comment]
Fixing review issues in https://github.com/emqx/emqx/pull/11743

**Pull request merged**